### ISSUE TITLE
Change labels of ZFIN and MGI genotypes to native syntax 

### ISSUE THREAD

**Issue by mbrush** [monarch-initiative/dipper#596]
MGI and ZFIN genotypes are still written using our proposed 'universal/normalized' genotype syntax we defined many years back. AGR curators wanted to see their native genotype syntax/labels that are displayed on their websites - as their users wouldn’t understand the universal format.  
  
Longer term, however, AGR curators agreed there may be value in creating a universal genotype format, with participation of all MODs in to define this.  Perhaps to show along side the native MOD syntax/label.  A medium term solution to help users understand what comprises genotypes in Monarch genotype records is showing a breakdown of genotype components according to the GENO model. This idea was proposed here several years back but not implemented.

**Comment by mbrush** [comment]
Hi @kshefchek.  Wanted to push again on this issue to change genotype labels for MGI and ZFIN genotypes, so they match the labels in the native source.  We are trying to get AGR buy in for GENO and Biolink, as a stepping stone to better relations between Monarch and AGR, and this is one of the things that wrankles them about our representation of their data.
 
Guessing we will just need to remove/replace parts of the ingest code that transforms the native label to our 'normalized' genotype syntax.  Happy to consult if this isn't straightforward, or there are questions about how the genotype labels should look. Ideally they would just appear exactly as on the MGI and ZFIN websites.
 
As for the other issue raised in the comments above concerning refactoring the IRIs we use for things, and the phenotype presentation issues - these may take a bit more work and can be done later.  But in the near term, can we just get the genotype labels fixed?  Thanks!

**Comment by kshefchek** [comment]
We're prioritizing this for the next release, @TomConlin can you tackle this for the ZFIN ingest?

**Comment by mbrush** [comment]
**Update**: I was browsing through old tickets and found [this one](https://github.com/monarch-initiative/dipper/issues/209#issuecomment-147534014) from Nicole W suggesting that the original MOD label for genotypes is captured as a *synonym* in the Dipper-generated ttl data.  Looking at the test ttl files [here](http://data.monarchinitiative.org/ttl/), this seems to indeed be the case. 
 
For **MGI**, the original labels are captured using the OIO:hasExactSynonym annotation property, and look like exactly what we want to have as our displayed label in the app.  e.g. the last line in the snippet below:

```
<http://www.informatics.jax.org/accession/MGI:2166381> a OBO:GENO_0000000,
        owl:NamedIndividual ;
    rdfs:label "Rbpj<tm1Kyo>/Rbpj<tm1Kyo> [involves: 129S2/SvPas * C57BL/6]" ;
    OBO:GENO_0000382 <https://monarchinitiative.org/.well-known/genid/mgivslckey1091> ;
    OBO:GENO_0000385 <https://monarchinitiative.org/.well-known/genid/mgistrainkey38138> ;
    dc:comment "_:mgigenotypekey87" ;
    OIO:hasExactSynonym "Rbpj<tm1Kyo>/Rbpj<tm1Kyo> [involves: 129S2/SvPas * C57BL/6]" .
```
The only thing we may want to consider here is to superscript the bracketed content < > in the labels we display on the web app - as described in #209.
 
------------------- 
 
The original **ZFIN** labels are also captured using  the oboInOwl:hasExactSynonym annotation property (note that the prefix is different from the one used in the MGI file), but may require a bit more processing to get to our desired display label. 
 
````
ZFIN:ZDB-GENO-080606-609 a OBO:GENO_0000000 ;
    rdfs:label "shha<tq252>/shha<+> [AB]" ;
    OBO:GENO_0000382 <https://monarchinitiative.org/.well-known/genid/ZDB-GENE-980526-166-ZDB-ALT-980413-636-WT> ;
    OBO:GENO_0000385 ZFIN:ZDB-GENO-960809-7 ;
    oboInOwl:hasExactSynonym "shha<sup>tq252/+</sup>",
        "tq252[1,U,U]AB" .
````
 
A few issues here:
1. Two synonyms are provided - the first is the one we want (`shha<sup>tq252/+</sup>`). The second is a shorthand that allows concise representation in a tsv file.
2. In our displayed label, HTML `<sup>` and `</sup>` tags in this label should be replaced with simple open angled brackets and close brackets, respectively. Longer term, we should consider code to present as a true superscript in the rendered label - as per #209.
3. The background is left off the first provided genotype synonym (e.g. `[AB]` is missing in the example above).  
     a. Note that the background does appear in the label on the [ZFIN webpage](https://zfin.org/action/genotype/view/ZDB-GENO-080606-609) for this genotype . 
     b. It may be worth checking to see if the latest version of the data provides this and we just need to update the parser in the Dipper ode to grab it.  Or we could write some code to parse the background out of the Monarch label that we currently create. 
4. ZFIN does not specify a background in their genotype label at all if the background is not specified  - e.g. see [here](https://zfin.org/action/genotype/view/ZDB-GENO-980410-268).  I think we will want to capture the background in the label even when unspecified - even if it means differing slightly from their original label.  Currently we use "[n.s.]" - but this is a bit opaque.  Thinking we should be as explicit as possible and use "[unspecified background]". 
5. The record for the ZFIN 'FISH' object related to this genotype has the same label, because there is not 'extrinsic/morpholino' component of the FISH  (see [here](https://zfin.org/ZDB-FISH-150901-24354)) - but I think that is fine.  
 
  
On a related note, we need to be sure that 'variant' labels are consistent with how the variant is resented within genotypes containing it.  Ticket #754 is relevant here - concerning removal of html tags from displayed variant/genotype labels in the app. This should be addressed concurrently. @jsutaddcoffee please review this ticket and lmk if it makes sense to do.

**Comment by justaddcoffee** [comment]
Thanks @mbrush! Some comments: 

> 1. Two synonyms are provided - the first is the one we want (shha<sup>tq252/+</sup>). The second is a shorthand that allows concise representation in a tsv file.

Is this ticket actually a UI issue then? I.e. do we just need to render the OIO:hasExactSynonym in the UI instead of changing the ingest? 

> 4. ZFIN does not specify a background in their genotype label at all if the background is not specified - e.g. see here. I think we will want to capture the background in the label even when unspecified - even if it means differing slightly from their original label.

This seems a bit at odds with the title of this ticket, although perhaps I'm misunderstanding. I.e. do we want to keep the label exactly as ingested, or add/subtract stuff from it?

**Comment by TomConlin** [comment]
where not specified the background use to be `AB`

**Comment by mbrush** [comment]
@justaddcoffee, re point 1, My inclination here is to change the ingest to capture the source MOD label as the rdfs:label, which is what the UI should show in the web app. That way if AGR folks look at the web app or the ttl dumps, they will find their native label as the primary label in our data. I would capture the Monarch universal syntax for the MGI and ZFIN genotypes (i.e. the *current* rdfs:label) as an exact synonym.  
- For MGI, this should be straightforward, as per my comments above.  
- For ZFIN we would have two exact synonym annotations (one for the tsv-based ZFIN name, e.g. `tbx392[1,U,U]AB` and one for the Monarch syntax name, e.g. `shha<tq252>/shha<+> [AB]`) - because there is only supposed to be one synonym per exactSynonym annotation (as opposed to a concatenated list) .  But this is not a big deal, if you want to concatenate.

re: point 4, the native label and syntax we aim to align with is what ZFIN shows on their web page to users.  As noted above, this includes the background (e.g. [here](https://zfin.org/action/genotype/view/ZDB-GENO-080606-609)).  So we should definitely append the background to ZFIN genotypes.

**Comment by kshefchek** [comment]
This has made its way to the app, eg https://monarchinitiative.org/genotype/ZFIN:ZDB-GENO-080606-609

I noticed that ZFIN fish labels are still different but rereading the ticket it seems we only targeted genotypes
